libraries/botbuilder-dialogs/src/dialogHelper.ts

```diff
@@ -84,9 +84,9 @@ export async function runDialog(
     }
 
     // Continue or start the dialog.
-    const result = await dialogContext.continueDialog();
+    let result = await dialogContext.continueDialog();
     if (result.status === DialogTurnStatus.empty) {
-        await dialogContext.beginDialog(dialog.id);
+        result = await dialogContext.beginDialog(dialog.id);
```

Comment:
Between what you [explained](https://github.com/microsoft/botbuilder-js/pull/3361#issue-585226650) and the [.NET implementation](https://github.com/microsoft/botbuilder-dotnet/blob/046e160829764c4653503b2ccf31acb8b46f2299/libraries/Microsoft.Bot.Builder.Dialogs/DialogExtensions.cs#L73-L76), this fix is spot on. 👍🏼

Reply:
Ah, yeah, I probably should've looked there.